pkg/cmd/codespace/ssh.go

```diff
@@ -147,6 +161,146 @@ func (a *App) SSH(ctx context.Context, sshArgs []string, opts sshOptions) (err e
 	}
 }
 
+func (a *App) printOpenSSHConfig(ctx context.Context, opts configOptions) error {
+	// Ensure all child tasks (e.g. port forwarding) terminate before return.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var err error
+	var codespaces []*api.Codespace
+	if opts.codespace == "" {
+		a.StartProgressIndicatorWithLabel("Fetching codespaces")
+		codespaces, err = a.apiClient.ListCodespaces(ctx, -1)
+		a.StopProgressIndicator()
+	} else {
+		var codespace *api.Codespace
+		codespace, err = getOrChooseCodespace(ctx, a.apiClient, opts.codespace)
+		codespaces = []*api.Codespace{codespace}
+	}
+	if err != nil {
+		return fmt.Errorf("error getting codespace info: %w", err)
+	}
+
+	t, err := template.New("ssh_config").Parse(`Host cs.{{.Name}}.{{.EscapedRef}}
+	User {{.SSHUser}}
+	ProxyCommand {{.GHExec}} cs ssh -c {{.Name}} --stdio
+	UserKnownHostsFile=/dev/null
+	StrictHostKeyChecking no
+	LogLevel quiet
+	ControlMaster auto
+
+`)
+	if err != nil {
+		return fmt.Errorf("error formatting template: %w", err)
+	}
+
+	ghexec, err := os.Executable()
+	if err != nil {
+		return err
+	}
+
+	// store a mapping of repository -> remote ssh username. This is
+	// necessary because the username can vary between codespaces, but
+	// since fetching it is slow, we store it here so we at least only do
+	// it once per repository.
+	sshUsers := map[string]string{}
+
+	for _, cs := range codespaces {
+
+		if cs.State != "Available" {
+			fmt.Fprintf(os.Stderr, "skipping unavailable codespace %s: %s\n", cs.Name, cs.State)
+			continue
+		}
+
+		sshUser, ok := sshUsers[cs.Repository.FullName]
+		if !ok {
+			codespace, err := a.apiClient.GetCodespace(ctx, cs.Name, true)
```

Comment:
This loop calls `GetCodespace` sequentially, which could be slow for lots of codespaces. I second @adonovan's suggestion that these lookups should ideally happen in parallel

Reply:
> However, can't this information be obtained for all codespaces at once through the list endpoint?

(which we already called earlier)

Reply:
Oh sorry, yes, you mean [this](https://github.com/cli/cli/pull/4915/commits/a05541f4ed924e9732144959790405bb1a63b39e). Apologies for being dense, I started this as proof of concept and was rushing a bit.